### ISSUE TITLE
expect_row_values_to_have_data_for_every_n_datepart fails when both a start and end date are passed

### ISSUE THREAD

**Issue by barberscott** [calogica/dbt-expectations#113]
When attempting to set up the `expect_row_values_to_have_data_for_every_n_datepart` test, if you pass in values for both `test_start_date` and `test_end_date` the test will fail with a compile error:

```
  001003 (42000): SQL compilation error:
  syntax error line 1 at position 0 unexpected 'None'.
```
This is due to the logic here:  https://github.com/calogica/dbt-expectations/blob/main/macros/schema_tests/distributional/expect_row_values_to_have_data_for_every_n_datepart.sql#L11-L28 -- the `{%- set dr = run_query(sql) -%}` call will be `None` if both date values are set.

The solution appears to be to move the logic in L26-28 inside the if statement.  This results in the sql not being run since there is no need to calculate the start/end dates in the database.

**Comment by clausherther** [comment]
Thanks for this @barberscott! I remember having some `None` issues at the time depending on where these values are set, so any fix here would need to be well tested. Would you be interested in submitting a PR for this one? No pressure if you can't, just want to double check. Otherwise I might have some time this weekend or next week for this.

**Comment by clausherther** [comment]
Closed via #115
